pkg/config/policy.go

```diff
@@ -0,0 +1,249 @@
+			}
+
+		case int32(cb.Policy_SIGNATURE):
+			sp := &cb.SignaturePolicyEnvelope{}
+			err := proto.Unmarshal(policy.Policy.Value, sp)
+			if err != nil {
+				return nil, err
+			}
+			rule, err := signatureMetaToString(sp)
+			if err != nil {
+				return nil, err
+			}
+
+			p[name] = &Policy{
+				Type: SignaturePolicyType,
+				Rule: rule,
+			}
+
+		default:
+			return nil, fmt.Errorf("unknown policy type: %v", policy.Policy.Type)
+		}
+	}
+	return p, nil
+}
+
+// implicitMetaToString converts a *cb.ImplicitMetaPolicy to a string representation.
+func implicitMetaToString(imp *cb.ImplicitMetaPolicy) (string, error) {
+	args := []string{}
+
+	switch imp.Rule {
+	case cb.ImplicitMetaPolicy_ANY:
+		args = append(args, cb.ImplicitMetaPolicy_ANY.String())
+	case cb.ImplicitMetaPolicy_ALL:
+		args = append(args, cb.ImplicitMetaPolicy_ALL.String())
+	case cb.ImplicitMetaPolicy_MAJORITY:
+		args = append(args, cb.ImplicitMetaPolicy_MAJORITY.String())
+	default:
+		return "", fmt.Errorf("unknown implicit meta policy rule type %v", imp.Rule)
+	}
+
+	args = append(args, imp.SubPolicy)
+
+	return strings.Join(args, " "), nil
+}
+
+// signatureMetaToString converts a *cb.SignaturePolicyEnvelope to a string representation.
+func signatureMetaToString(sig *cb.SignaturePolicyEnvelope) (string, error) {
+	roles := []string{}
+	for _, id := range sig.Identities {
+		role, err := mspPrincipalToString(id)
+		if err != nil {
+			return "", err
+		}
+		roles = append(roles, role)
+	}
+	return signaturePolicyToString(sig.Rule, roles)
+}
+
+// mspPrincipalToString converts a *mb.MSPPrincipal to a string representation.
+func mspPrincipalToString(principal *mb.MSPPrincipal) (string, error) {
+	var res strings.Builder
+
+	switch principal.PrincipalClassification {
+	case mb.MSPPrincipal_ROLE:
+		role := &mb.MSPRole{}
+		err := proto.Unmarshal(principal.Principal, role)
+		if err != nil {
+			return "", err
+		}
+		res.WriteString("'")
+		res.WriteString(role.MspIdentifier)
+		res.WriteString(".")
+		res.WriteString(strings.ToLower(role.Role.String()))
+		res.WriteString("'")
+
+		return res.String(), nil
+		// TODO: currently fabric only support string to principle convertion for
+		// type ROLE. Implement MSPPrinciple to String for types ORGANIZATION_UNIT,
+		// IDENTITY, ANONYMITY, and GOMBINED once we have support from fabric.
+	case mb.MSPPrincipal_ORGANIZATION_UNIT:
+		return "", nil
+	case mb.MSPPrincipal_IDENTITY:
+		return "", nil
+	case mb.MSPPrincipal_ANONYMITY:
+		return "", nil
+	case mb.MSPPrincipal_COMBINED:
+		return "", nil
+	default:
+		return "", fmt.Errorf("unknown MSP principal classiciation %v", principal.PrincipalClassification)
+	}
+}
+
+// signaturePolicyToString recursively converts a *cb.SignaturePolicy to a
+// string representation.
+func signaturePolicyToString(sig *cb.SignaturePolicy, IDs []string) (string, error) {
+	switch sig.Type.(type) {
+	case *cb.SignaturePolicy_NOutOf_:
+		nOutOf := sig.GetNOutOf()
+		policies := []string{}
```

Comment:
nit: Another empty slice;  If I’m reading this correctly, it basically gets initialized on line 224 with the value of nOutOf.N.

Reply:
done.